thriftsimple/thriftsimple.go

```diff
@@ -0,0 +1,160 @@
+package thriftsimple
+
+import (
+	"context"
+	"errors"
+	"io/ioutil"
+	"time"
+
+	"github.com/apache/thrift/lib/go/thrift"
+	"github.com/getsentry/raven-go"
+	"github.com/reddit/baseplate.go/edgecontext"
+	"github.com/reddit/baseplate.go/log"
+	"github.com/reddit/baseplate.go/metricsbp"
+	"github.com/reddit/baseplate.go/secrets"
+	"github.com/reddit/baseplate.go/thriftbp"
+	"github.com/reddit/baseplate.go/tracing"
+	"gopkg.in/yaml.v2"
+)
+
+type BaseplateServerConfig struct {
+	Addr string
+
+	Timeout time.Duration
+
+	Debug bool
+
+	Metrics struct {
```

Comment:
Should these be separate `config` structs that we define in the packages and embed in this struct?

One potential benefit of that is we can change those `itit*` functions to exported `Init*` functions that use those specific configs rather than the top-level `BaseplateServerConfig`.

Reply:
That sounds quite nice, and a wider-ranging refactor.

Reply:
Let's do this in a follow-up patch.  That should effectively be an internal refactor, as the inits are private now.